server/server.go

```diff
@@ -3892,7 +3892,10 @@ func (s *StanServer) removeAllNonDurableSubscribers(client *client) {
 	subs := client.subs
 	clientID := client.info.ID
 	client.RUnlock()
-	var storesToFlush map[string]stores.SubStore
+	var (
+		storesToFlush map[string]stores.SubStore
```

Comment:
If not critical performance wise just create empty storesToFlush like you do with the map below. Then you can eliminate nil checks.